src/main/java/org/primeframework/mvc/message/l10n/ResourceBundleMessageProvider.java

```diff
@@ -120,21 +123,26 @@ protected Queue<String> determineBundles(String bundle) {
 
   /**
    * Finds the message in a resource bundle using the search method described in the class comment.
+   * If the action was annotated with a {@link AlternateMessageResources} annotation
+   * and the message was not found with the request's action, that action will be searched as well.
    *
    * @param actionInvocation The action invocation.
    * @param key              The key of the message.
    * @return The message or null if it doesn't exist.
    */
   protected String findMessage(ActionInvocation actionInvocation, String key) {
-    String uri = actionInvocation.actionURI;
-    Queue<String> names = determineBundles(uri);
-    for (String name : names) {
-      try {
-        ResourceBundle rb = ResourceBundle.getBundle(name, localeProvider.get(), control);
-        return rb.getString(key);
-      } catch (MissingResourceException ignore) {
-        // Ignore and check the next bundle
-      }
+    String message = findMessage(actionInvocation.actionURI, key);
+    if (message != null) {
+      return message;
+    }
+
+    ActionConfiguration config = actionInvocation.configuration;
+    if (config != null) {
+      return config.fallbackActionMessageURIs.stream()
+                                             .map(uri -> findMessage(uri, key))
+                                             .filter(Objects::nonNull)
+                                             .findFirst()
+                                             .orElse(null);
```

Comment:
Is it possible to have a `null` value here? If we have an `ActionInvocation` object passed in that is non-null, assuming that means we always have an action configuration.

Reply:
The `config` is null only in some of the tests. I don't think in reality. That's what I meant by "I'd rather bake in the assumption that we always have an action config. But it was much simpler test wise to do this" above. But probably the right answer to fix the tests.

Reply:
Fixed.

Reply:
ah.. got it. Ok. I can live with that.